xs2a-adapter-service-api-adapter/src/main/java/de/adorsys/xs2a/adapter/http/ApacheHttpClient.java

```diff
@@ -120,12 +130,49 @@ public String content(Request.Builder requestBuilder) {
             InputStream content = entity != null ? entity.getContent() : EmptyInputStream.INSTANCE;
 
             T responseBody = responseHandler.apply(statusCode, content, responseHeaders);
+            logResponse(response, responseHeadersMap, responseBody);
             return new Response<>(statusCode, responseBody, responseHeaders);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    private <T> void logResponse(
+        CloseableHttpResponse response,
+        Map<String, String> headers,
+        T responseBody
+    ) {
+        logger.debug("<-- {}", response.getStatusLine());
+        headers.keySet().forEach(
+            key -> logger.debug("<-- {}: {}", key, headers.get(key))
+        );
+        logger.debug("<--");
+        Optional<String> contentType = Optional.ofNullable(
+            headers.get(ResponseHeaders.CONTENT_TYPE)
+        );
+        if (contentType.isPresent() && contentType.get().startsWith("application/json")) {
+            try {
+                logger.debug(RESPONSE_BODY, new ObjectMapper().writeValueAsString(responseBody));
+            } catch (JsonProcessingException e) {
+                logger.debug(RESPONSE_BODY, responseBody);
+            }
+        } else {
+            logger.debug(RESPONSE_BODY, "*******");
+        }
+    }
+
+    private void logRequest(HttpUriRequest request, Map<String, String> headers) {
+        logger.debug("--> {}", request.getRequestLine());
+        headers.keySet().forEach(
+            headerName -> logger.debug("--> {}: {}", headerName, headers.get(headerName))
+        );
+        logger.debug("-->");
+        String requestBody = getContent(request);
+        if (StringUri.isNotBlank(requestBody)) {
```

Comment:
This check doesn't look necessary to me.

Reply:
thanks, agree